### PR TITLE
Snowflake connector

### LINKED ISSUE #3394: Runtime: Snowflake connector for sources
The implementation should take an approach similar to our Athena and BigQuery connectors.

### PR DESCRIPTION
**Pull request by esevastyanov**
Closes #3394

### CODE REVIEW

File: web-common/src/features/sources/modal/submitRemoteSourceForm.ts
Comment (k-anshul) [inline web-common/src/features/sources/modal/submitRemoteSourceForm.ts:67]: I think this should not be required.

File: web-common/src/features/sources/modal/yupSchemas.ts
Comment (k-anshul) [inline web-common/src/features/sources/modal/yupSchemas.ts:117]: This should be `dsn`?

File: runtime/drivers/snowflake/sql_store.go
Comment (k-anshul) [inline runtime/drivers/snowflake/sql_store.go:78]: This error should be handled

File: runtime/drivers/snowflake/sql_store.go
Comment (k-anshul) [inline runtime/drivers/snowflake/sql_store.go:72]: Can `rows` be `nil` if `conn.Raw` returns error ?

File: runtime/drivers/snowflake/sql_store.go
Comment (k-anshul) [inline runtime/drivers/snowflake/sql_store.go:55]: This should return error instead of `fatal log`

File: runtime/drivers/snowflake/sql_store.go
Comment (k-anshul) [inline runtime/drivers/snowflake/sql_store.go:75]: This should return error instead of `log.Fatalf` which exits the program.

File: runtime/drivers/snowflake/sql_store.go
Comment (k-anshul) [inline runtime/drivers/snowflake/sql_store.go:146]: I think this should return some error like `no results found for the query`. Check postgres/bigquery.
Also this can be handled in `QueryAsFiles` itself.

**Review by k-anshul — CHANGES_REQUESTED** [review]

File: runtime/drivers/bigquery/arrow.go
Comment (esevastyanov) [reply runtime/drivers/bigquery/arrow.go:11]: Yes, otherwise it leads to
```
panic: proto: file "Flight.proto" is already registered
        previously from: "github.com/apache/arrow/go/v12/arrow/flight/internal/flight"
        currently from:  "github.com/apache/arrow/go/v13/arrow/flight/internal/flight"
See https://protobuf.dev/reference/go/faq#namespace-conflict
```
Snowflake uses v12 and I don't see any PRs or issues in a corresponding repo to move to v13 or even v14.

Comment (k-anshul) [reply runtime/drivers/bigquery/arrow.go:11]: Okay we will need to solve this soon since we need to upgrade to v14. I see this open [issue](https://github.com/apache/arrow/issues/37353) on arrow as well.
I guess for now we can merge this and try to find alternatives once we upgrade to v14 unless you or @begelundmuller have other easy solutions.

**Review by k-anshul — APPROVED** [review]
LGTM. 
Please also see the comment here : https://github.com/rilldata/rill/pull/3555#discussion_r1407577252

File: runtime/drivers/snowflake/snowflake.go
Comment (begelundmuller) [inline runtime/drivers/snowflake/snowflake.go:30]:
```suggestion
			DisplayName: "Snowflake Connection String",
```

**Review by begelundmuller — APPROVED** [review]
LGTM!

**Comment by begelundmuller** [comment]
The Arrow library discrepancy is a bummer – @esevastyanov can you file an issue (or send a PR) on `gosnowflake` to upgrade it?

If they don't want to upgrade, we might need to keep a fork that uses a different Arrow version.

But let's not let that block this PR.

**Comment by esevastyanov** [comment]
@begelundmuller @k-anshul I raised a [PR](https://github.com/snowflakedb/gosnowflake/pull/977) to upgrade apache/arrow/go in gosnowflake. But the next release is in [January](https://github.com/snowflakedb/gosnowflake/pull/977#issuecomment-1831867289)

**Pull request merged**
